From 12559a2581e8517f35c142b04bd7ed7830307b9d Mon Sep 17 00:00:00 2001
From: Mark Salter <msalter@redhat.com>
Date: Thu, 7 Jan 2016 20:53:41 +0100
Subject: arm-efi: Reduce timer event frequency by 10

Timer event to keep grub msec counter was running at 1000HZ. This was too
fast for UEFI timer driver and resulted in a 10x slowdown in grub time
versus wallclock. Reduce the timer event frequency and increase tick
increment accordingly to keep better time.

Origin: upstream, http://git.savannah.gnu.org/cgit/grub.git/commit/?id=7eb27a49e92623587f400ff58de3e9b1a31ba894
Last-Update: 2016-03-04

Patch-Name: arm-efi-Reduce-timer-event-frequency-by-10.patch
---
 grub-core/kern/arm/efi/init.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grub-core/kern/arm/efi/init.c b/grub-core/kern/arm/efi/init.c
index a6ae034..0c17d83 100644
--- a/grub-core/kern/arm/efi/init.c
+++ b/grub-core/kern/arm/efi/init.c
@@ -38,7 +38,7 @@ static void
 increment_timer (grub_efi_event_t event __attribute__ ((unused)),
 		 void *context __attribute__ ((unused)))
 {
-  tmr++;
+  tmr += 10;
 }
 
 void
@@ -52,7 +52,7 @@ grub_machine_init (void)
 
   efi_call_5 (b->create_event, GRUB_EFI_EVT_TIMER | GRUB_EFI_EVT_NOTIFY_SIGNAL,
 	      GRUB_EFI_TPL_CALLBACK, increment_timer, NULL, &tmr_evt);
-  efi_call_3 (b->set_timer, tmr_evt, GRUB_EFI_TIMER_PERIODIC, 10000);
+  efi_call_3 (b->set_timer, tmr_evt, GRUB_EFI_TIMER_PERIODIC, 100000);
 
   grub_install_get_time_ms (grub_efi_get_time_ms);
 }
